pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -264,3 +265,50 @@ func aggregateStatefulSetStatus(object *unstructured.Unstructured, aggregatedSta
 
 	return helper.ToUnstructured(statefulSet)
 }
+
+func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
+	pod, err := helper.ConvertToPod(object)
+	if err != nil {
+		return nil, err
+	}
+
+	newStatus := &corev1.PodStatus{}
+	newStatus.ContainerStatuses = make([]corev1.ContainerStatus, 0)
+	readySum := 0
+	containerSum := 0
+	for _, item := range aggregatedStatusItems {
+		if item.Status == nil {
+			continue
+		}
+
+		temp := &corev1.PodStatus{}
+		if err = json.Unmarshal(item.Status.Raw, temp); err != nil {
+			return nil, err
+		}
+
+		for _, containerStatus := range temp.ContainerStatuses {
+			tempStatus := containerStatus
+			newStatus.ContainerStatuses = append(newStatus.ContainerStatuses, tempStatus)
+			if tempStatus.Ready {
+				readySum++
+			}
+			containerSum++
+		}
+		klog.V(3).Infof("Grab pod(%s/%s) status from cluster(%s), phase: %s", pod.Namespace,
+			pod.Name, item.ClusterName, temp.Phase)
+	}
+
+	if reflect.DeepEqual(pod.Status, *newStatus) {
+		klog.V(3).Infof("ignore update pod(%s/%s) status as up to date", pod.Namespace, pod.Name)
+		return object, nil
+	}
```

Comment:
This logic maybe needs to move back after `newStatus.Phase` assignment.

Reply:
> This logic maybe needs to move back after `newStatus.Phase` assignment.

Okey. It will look more reasonable.